services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,17 +1,49 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
-import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
+import cromwell.pipeline.datastorage.dto.{
+  FileParameter,
+  ProjectConfiguration,
+  ProjectFileConfiguration,
+  ProjectId,
+  TypedValue
+}
 
+import java.nio.file.Path
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
-    repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
-      .map(_.toString)
+  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] = {
+    def updateConfiguration(configuration: ProjectConfiguration) =
+      repository.updateOne(toDocument(configuration), "projectId", projectConfiguration.projectId.value).map(_.toString)
+
+    def toMap(config: ProjectFileConfiguration): Map[Path, Map[String, TypedValue]] =
+      Map(
+        config.path -> config.inputs.flatMap(fileParameter => Map(fileParameter.name -> fileParameter.typedValue)).toMap
+      )
+
+    def toProjectFileConfiguration(path: Path, inputs: Map[String, TypedValue]): ProjectFileConfiguration =
+      ProjectFileConfiguration(path, inputs.flatMap {
+        case (name, typedValue) => List(FileParameter(name, typedValue))
+      }.toList)
```

Comment:
```suggestion
      ProjectFileConfiguration(path, inputs.map((FileParameter.apply _).tupled).toList)
```
Now once it's clear what you want to do, we can simplify it